app/src/main/java/org/glucosio/android/activity/FreestyleLibreActivity.java

```diff
@@ -176,7 +176,7 @@ private void showReadingLayout() {
 
         // Apply values in TextViews
         // TODO: Add check for mmol/L
-        readingTextView.setText(mResult.history.get(mResult.history.size()-1).glucose(false));
+        readingTextView.setText(mResult.trend.get(0).glucose(false));
 
         new Runnable() {
```

Comment:
didn't touch that yet and didn't check what is really needed, but there is imho a lot of code in the FSL-parts, we don't need/use. If it works good for a few people like this, i would start to clean up the code..